mobile/src/reducers/calendar/approval.epics.ts

```diff
@@ -0,0 +1,65 @@
+/******************************************************************************
+ * Copyright (c) Arcadia, Inc. All rights reserved.
+ ******************************************************************************/
+
+import { ActionsObservable, ofType, StateObservable } from 'redux-observable';
+import { filter, flatMap, map, mergeMap, tap } from 'rxjs/operators';
+import { handleHttpErrors } from '../errors/errors.epics';
+import { deserializeArray } from 'santee-dcts';
+import { ApprovalActionType, Approve, approveFinished, LoadApprovals, loadApprovalsFinished } from './approval.action';
+import { AppState, DependenciesContainer } from '../app.reducer';
+import { Approval } from './approval.model';
+import { Map as ImmutableMap, Set } from 'immutable';
+import { forkJoin } from 'rxjs';
+
+//----------------------------------------------------------------------------
+export const loadApprovals$ = (action$: ActionsObservable<LoadApprovals>, _: StateObservable<AppState>, deps: DependenciesContainer) => action$.pipe(
+    ofType(ApprovalActionType.loadApprovals),
+    mergeMap(action => {
+        const employeeId = action.employeeId;
+        const eventIds = Set(action.eventIds);
+        const requests = eventIds.toArray().map(eventId => {
+            return deps.apiClient.getJSON(`/employees/${employeeId}/events/${eventId}/approvals`).pipe(
+                map(obj => deserializeArray(obj as any, Approval)),
+                tap(approvals => {
+                    approvals.forEach(approval => { approval.eventId = eventId; });
+                }),
+                map(approvals => {
+                    return { eventId: eventId, approvals: approvals };
+                }),
+                handleHttpErrors(),
+            );
+        });
+        return forkJoin(requests).pipe(
+            map(approvalsForEventId => {
+                const approvals = new Map();
+                approvalsForEventId.forEach(x => {
+                    approvals.set(x.eventId, x.approvals);
+                });
+                return { employeeId: employeeId, approvals: ImmutableMap(approvals) };
+            }),
+        );
+    }),
+    map(holder => loadApprovalsFinished(holder.employeeId, holder.approvals)),
+);
+
+//----------------------------------------------------------------------------
+export const approve$ = (action$: ActionsObservable<Approve>, state$: StateObservable<AppState>, deps: DependenciesContainer) => action$.pipe(
+    ofType(ApprovalActionType.approve),
+    filter(() => !!state$.value.userInfo && !!state$.value.userInfo.employeeId),
+    flatMap(action => {
+        const body = {
+            approverId: state$.value.userInfo!.employeeId!,
```

Comment:
I suggest to pass approverId in action instead. First, it's natural, you'd always know who approved. Second, you won't need this filter.

Reply:
ec889399498e866df3b7b3bef8816294f209f393